File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
    IDictionary<int, int> map = new Dictionary<int, int>();
    for (int i = 0; i < nums.Length; i++)
    {
        if (map.ContainsKey(target - nums[i]))
            return [ map[target - nums[i]], i ];
        if (!map.ContainsKey(nums[i]))
        map.Add(nums[i], i);
    }
    return [];
        }
}

File: 0001-two-sum/0001-two-sum.cpp
#include <unordered_map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
         map<int, int> indices;
        for (int i = 0; i < nums.size(); i++) {
            if (indices.find(target - nums[i]) != indices.end()) {
                return {indices[target - nums[i]], i};
            }
            indices[nums[i]] = i;
        }
        return {};
    }
};